my-vue3-app/src/store/reports/actions.ts

```diff
@@ -1,21 +1,56 @@
 import { ActionContext } from 'vuex';
 import api from '@/services/api';
-import { Pagination } from './state';
-import { SET_DATA, SET_PAGINATION, SET_LOADING } from './mutations';
+import {
+  PAGINATION, Pagination, Reports, REPORTS, State,
+} from './state';
+import {
+  ADD_DATA, SET_PAGINATION, SET_LOADING,
+} from './mutations';
 
 export const FETCH_REPORTS = 'load_reports';
 
+const applyFilterSortToReports = (
+  pagination: Pick<Pagination, 'order'>,
+  reports: Reports,
+): Array<string> => Object
+  .keys(reports)
+  // .map((id) => reports[id]);
+  // filter
+  // .map((report) => report.id);
+  .sort((a, b) => a.localeCompare(b) * pagination.order.id);
+
 export default {
-  async [FETCH_REPORTS]({ commit }: ActionContext<unknown, unknown>, payload: Pagination) {
+  async [FETCH_REPORTS](
+    { commit, state }: ActionContext<State, unknown>,
+    desired: Omit<Pagination, 'viewIds'|'ids'>,
```

Comment:
what is Omit?

Reply:
Same as Pick, but opposite :)
It is TS data generic type
https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys